### DISCUSSION TITLE
Can I use capacitor with a Rails React web app?

### DISCUSSION THREAD

**Discussion by echan00** [ionic-team/capacitor#5025] [Q&A]
Hi, I'm hoping to use Capacitor to create an iOS app out of my rails react app (helladoge.com).  From what I gather, this ought to be very possible but am having trouble with setup:

What is this `webDir` with `index.html`?  More specifically what is this `index.html` supposed to contain? 
I'm trying to figure out the equivalent in Rails React (since there are many), is this the index.html my domain root points to?  

For those who are not familiar with Rails (or Rails React), Rails uses a `view/layouts/application.html` template for all pages. Inside this layout html file, there is a <%= yield %> ruby tag where each different route/page will insert its content into the body. Sometimes the content of each route/page will render (e.g embed) additional content from another Html file.

I guess there isn't a primary main index.html file in a Rails React web application. I would appreciate any ideas/suggestions or if anyone could point me towards any references.

**Comment by mlynch** [comment]
Hi @echan00, yes, you can use Rails with Capacitor, but there are some nuances here. First, Capacitor's `@capacitor/core` library can be used in any modern JS app, such as one loaded from a request to a server running rails. This would provide some APIs to make building PWAs a bit nicer.

However, if you'd like to then build iOS and Android apps, rails can't just run in those environments because they are not true server environments. Instead, you'd need to either save a snapshot of the rendered HTML for the rails site (which is probably not very useful), or use the remote `server.url` config in Capacitor to load the app from your rails server.

The last option is probably what you were expecting, but there are some downsides to this. Apple, for example, is strict about UX of mobile apps, and if they think your app is simply loading a remote website, they might not accept your app. We have been pushing people away from this config option but so many people want to use it that it's really at your own risk given that it falls into an app store UX gray area and depends heavily on the experience you build: https://capacitorjs.com/docs/config

Hope that helps

**Comment by echan00** [comment]
Hi @mlynch thanks for the detailed response. To make sure I understand:

There are two options: 
1) Building a PWA as documented here (https://capacitorjs.com/docs/web) 
2) Follow the same instructions (https://capacitorjs.com/docs/getting-started) but try the following to get around the index.html requirement:
i) Save a snapshot of the rendered HTML from the rails site
or ii) Use the remote server.url config in Capacitor to load the app

You say Option 2 ("the last option") is not suggested because Apple does not like apps that are essentially a browser rendering a website. Correct?  Or you are not suggesting to go with Option 1?

-----

EDIT:
Upon re-reading your comment a few more times, I believe you are suggesting:
- I can use @capacitor/core for my React app
- Then, i) save a snapshot of the rendered HTML for the rails site OR ii) use the remote server.url config in Capacitor to load the app from your rails server (but not suggested)

Correct?
